polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1797,12 +1799,16 @@ mlir::Type CodeGenTypes::getPointerOrMemRefType(mlir::Type Ty,
                                                 unsigned AddressSpace,
                                                 bool IsAlloc) const {
   auto ST = dyn_cast<mlir::LLVM::LLVMStructType>(Ty);
+  auto PolygeistST = dyn_cast<polygeist::StructType>(Ty);
 
   bool IsSYCLType = mlir::sycl::isSYCLType(Ty);
   if (ST)
     IsSYCLType |= any_of(ST.getBody(), mlir::sycl::isSYCLType);
+  if (PolygeistST)
+    IsSYCLType |= any_of(PolygeistST.getBody(), mlir::sycl::isSYCLType);
 
-  if (!ST || IsSYCLType)
+  // TODO: Use MemRefType for polygeist::StructType.
+  if (!(ST || PolygeistST) || IsSYCLType)
     return mlir::MemRefType::get(IsAlloc ? 1 : ShapedType::kDynamic, Ty, {},
                                  AddressSpace);
   return getPointerType(Ty, AddressSpace);
```

Comment:
I think a typeswitch can greatly simplify this. Also, why don't we use `memref` for `polygeist.struct` right away?

Reply:
Compile failures for using `memref`? I think I don't follow you, sorry

Reply:
right, by returning `memref<?x!polygeist.struct>` in this function, we hit some compile time failures, which need to be look at.